### DISCUSSION TITLE
Flowers with transparent background

### DISCUSSION THREAD

**Discussion by quasar136a8** [lllyasviel/Fooocus#2699] [Q&A] [answered]
Hi, I have a question, is it possible to create, for example, flowers with a transparent background in png format, for example tall roses
![family-tree-rose-flowerpot-garden-roses-bonsai-plants-seed-ornamental-plant-png-clipart](https://github.com/lllyasviel/Fooocus/assets/111929719/619debce-5be9-4d5e-962f-ece377619249)

**Comment by ILikeToasters** [comment, marked as answer]
No you can't do transparent backgrounds with Fooocus. Your best bet would be to try to do something like white background and remove with another tool. I use Gimp myself for that.

**Reply by mashb1t** [reply]
at least not yet, until LayeredDiffusion has been implemented, if it ever will be.

**Reply by vecuccio** [reply]
@mashb1t How do those AI powered online background remover websites work,that take your photo and spit out a transparent PNG ?

**Reply by mashb1t** [reply]
@vecuccio mostly by using detection models. You can find great examples and docs how they (in this case u2net) work here:
- https://github.com/danielgatis/rembg?tab=readme-ov-file#references
- https://arxiv.org/pdf/2005.09007.pdf
- https://github.com/xuebinqin/U-2-Net

**Reply by vecuccio** [reply]
Thanks. REMBG looks excellent. However it is still in a form where it can only be used by codies, as there's no package to download.
Same with U-2.net, though it has a link to Sliueta.me - which does not seem to work with Firefox or Brave, as the server is permanently stuck in a restart loop. Or maybe it genuinely has been restarting for 15 minutes.
It's a real shame that noone built a GUI for either of them like you and Illyasviel did with Fooocus.

The REMBG github has a section called "Models", with download links. As far as I can tell, alot of those are auto downloaded by Fooocus into the models\inpaint folder. But there are some I don't have that I would like to try. Are all of these models compatible with Fooocus ?

**Reply by mashb1t** [reply]
@vecuccio as you've mentioned Fooocus uses rembg for background removal except for SAM, see https://github.com/lllyasviel/Fooocus/blob/d7439b2d6004d50a0fda19108603a8d1941a185e/extras/inpaint_mask.py#L62. All other models should also be compatible, given the version of rembg used in Fooocus supports them.

**Reply by mashb1t** [reply]
@vecuccio you can find an online GUI for rembg on huggingface here: https://huggingface.co/spaces/KenjieDec/RemBG
